src/main/java/com/powsybl/openloadflow/sa/OpenSecurityAnalysis.java

```diff
@@ -302,6 +346,11 @@ private PostContingencyResult runPostContingencySimulation(LfNetwork network, Ac
                 network.getBranches().stream().filter(b -> !lfContingency.getBranches().contains(b)),
                 network.getBuses().stream().filter(b -> !lfContingency.getBuses().contains(b)),
                 postContingencyLimitViolations);
+            StateMonitor stateMonitor = monitors.get(lfContingency.getContingency().getId());
+            if (stateMonitor != null && (stateMonitor.getContingencyContext().getContextType() == ContingencyContextType.ALL
+                || stateMonitor.getContingencyContext().getContextType() == ContingencyContextType.SPECIFIC)) {
+                addMonitorInfos(network, stateMonitor, branchResults::add, busResults::add, threeWindingsTransformerResults::add);
```

Comment:
Maybe there is something to do here to filter out buses and branches which are disconnected by the contingency, see filters above in the identification of violations.

Not 100% sure about that : @annetill and @geofjamg do you know if there is already a test case covering that ?

Reply:
yes, use `isDisabled` for buses. For branches, the method has to be created.